docs/manual/src/internals/lifting_and_lowering.md

```diff
@@ -13,11 +13,55 @@ receiving side "***lifts***" that primitive type into its own language-specific
 data type.
 
 Lifting and lowering simple types such as integers is done by directly casting the
-value to and from an appropriate type. For complex types such as optionals and
-records we currently implement lifting and lowering by serializing into a byte
-buffer, but this is an implementation detail that may change in future. (See
-[ADR-0002](https://github.com/mozilla/uniffi-rs/blob/main/docs/adr/0002-serialize-complex-datatypes.md) for the reasoning
-behind this choice.)
+value to and from an appropriate type. Complex types such as optionals and
+records are lowered to a byte buffer called a `RustBuffer` internally.
```

Comment:
Maybe replace "Complex types such as optionals and records are lowered" with something like "Non-trivial types such as Strings, Optionals, Records, etc"? This ties in better with your String example in the ffi traits docs. Not that happy with "non-trivial" either, but it's mildly better than "complex"? Even "primitive" isn't ideal because (a) String is a primitive and it doesn't handle pointers. Whatever you decide :)

Reply:
Yeah, it's always hard to come up with words for this.  But I like "non-trivial".